frontend/packages/kubevirt-plugin/src/components/modals/disk-modal/disk-modal.tsx

```diff
@@ -167,8 +168,17 @@ export const DiskModal = withHandlePromise((props: DiskModalProps) => {
         setAccessMode(getDefaultSCAccessModes(storageClassConfigMap)[0]);
```

Comment:
we should pass SC to get the correct access modes and volume modes

Reply:
hm, now I see we have a very similar logic in onStorageClassNameChanged, but sligtly different. Maybe we could extract the logic on lines 287-294 into a function?

Reply:
a - using the sc now to pick the access and volume
b - the logic is far enough to make merging the onChange and onInit code look forced

Reply:
ok